streamalert/shared/config.py

```diff
@@ -218,9 +218,37 @@ def load_config(conf_dir='conf/', exclude=None, include=None, validate=True):
     if validate:
         _validate_config(config)
 
+    if config.get('logs'):
+        config['logs'] = _sanitize_logs_name(config['logs'])
+
+    if (config.get('global')
+            and config['global'].get('infrastructure')
+            and config['global']['infrastructure'].get('firehose')
+            and config['global']['infrastructure']['firehose'].get('enabled_logs')):
+        config['global']['infrastructure']['firehose']['enabled_logs'] = _sanitize_logs_name(
+            config['global']['infrastructure']['firehose']['enabled_logs']
+        )
+
     return config
 
 
+def _sanitize_logs_name(config):
```

Comment:
does theis `_sanitize_logs_name` function actually _replace_ the value within a person's config to be written back out to disk??

if so, I don't think we should do this... this could mess up rules, etc that use specific logs. I think and "transformation" for log name --> table name + firehose name should be done by _us_ under the hood

Reply:
I don't think `_sanitize_logs_name` change a person's config. In my staging environment, I don't see anything changed under `conf/`. 
Totally agree with you that the changes should be done by us under the hood. I had iterated few solutions and find current change is the best. But I may miss something. I will look at rule and classifier code again today.